Assets/Mirror/Runtime/NetworkMessage.cs

```diff
@@ -8,7 +8,8 @@ public struct NetworkMessage
 
         public TMsg ReadMessage<TMsg>() where TMsg : IMessageBase, new()
         {
-            TMsg msg = new TMsg();
+            TMsg msg = typeof(TMsg).IsValueType ? default(TMsg) : new TMsg();
```

Comment:
a comment about this wizardry would be useful too :)

Reply:
fixed